notes.md

```diff
@@ -0,0 +1,115 @@
+- It can determine the game is over and inform the user providing their final score
+- Edge cases:
+* Scores given are negatives/above 10
+
+User Stories
+
+As a player
+So I can enjoy a game of bowling
+I want to be able to enter my score
+
+As a player
+So I can enjoy a game of bowling
+I want my score to be returned to me after each frame
+
+As a player
+So I can know my final score
+I want my score to be saved
+
+As a player
+So I can have my correct score
+I want to be given bonuses for bowling strikes and spares
+
+As a player
+So I can enjoy the full game
+I want to be able to have an extra bowl in the 10th frame if applicable
+
+As a player
+So I can enjoy a game of bowling
+I want to know when the game is over
+
+Best way to do the scoring for strikes and spares is to check after each bowl if a strike or spare has been bowled the previous frame.
+Also check if a strike has been bowled the previous frame if one was bowled the frame before. Bonus can only up to 20 and then 30.
+
+Plan of action:
+
+TDD
+Fight the easy fights to start with:
+
+- Create bowling class
+- Create function that takes a user input
+Frame
+- It then saves this user input into a frame array
+- If the score input is 10 then frameStatus = strike
+- If the score input is less than 10, prompt for second bowl
+- If total for frame is 10 frameStatus = spare
+- Otherwise totalScore is pushes into Frame array
+- frameScore and totalScore are returned to user
+- Currently have 2 bowl methods handling the scoring of each bowl.
+- Thinking add a frame method which handles the directing. I.e If bowl1 != 'strike' bowl 2
+else change prevFrame status to 1xstrike.
+
+Bonuses
+- When the first ball if NOT a strike:
+- Check if previous ball was a spare, if so add first round score to 10 for previous frame and push that into scores array
+
+- Check if 2x strike, if so add first bowl score to 20 and push into 2x previous frame array - come back to
+
+- if 2x strike and spare add 20 + first bowl to original strike frame
+and 20 to second
+-
+- if 2xstrike and not spare or strike add 20 + first score to 2x previous
+- add 10 + framescore
+
+- check if prevFrame is a strike and add 10 to frame score
+- If is a strike:
+- Check if previous ball was a spare and add 20 to previous frame
+- Check if previous ball was a strike, if so change frameStatus to 2x strike
+- Check if previous ball is 2x strike, if so add 30 to 2x previous frame array
+
+- framestatus is spare and prevframe is strike, 10+10
+
+10th Frame
+- If first ball NOT a strike
+- Check bonuses
+- Second bowl is NOT a spare
+- Check bonuses
+- Game ends  - score calculated and returned
+- Second bowl IS a spare
+- check bonuses
+- Extra bowl and spare and next bowl score combined
+
+- If first ball IS a strike
+- Check bonuses
+- If next ball is NOT a strike
+- Check bonuses
+- Bowl last ball and add up
+- IF next ball IS a strike
+- check bonuses
+- Bowl last ball and add up
```

Comment:
Nice way to break down the problem - did you end up making a flow diagram similar to Stu, would be easier to follow